src/fidesops/main.py

```diff
@@ -40,6 +42,75 @@
         allow_headers=["*"],
     )
 
+
+@app.middleware("http")
+async def dispatch_log_request(request: Request, call_next: Callable) -> Response:
+    fides_source: Optional[str] = request.headers.get("X-Fides-Source")
+    now: datetime = datetime.now(tz=timezone.utc)
+    endpoint = f"{request.method}: {request.url.path}"
+
+    try:
+        response = await call_next(request)
+        # HTTPExceptions are considered a handled err by default so are not thrown here.
+        # Accepted workaround is to inspect status code of response.
+        # More context- https://github.com/tiangolo/fastapi/issues/1840
+        try:
+            if response.status_code >= 400:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    "HTTPException",
+                )
+            else:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    None,
+                )
+        except Exception as exc:
+            # always continue if something went wrong with analytics
+            logger.warning(
+                f"Analytics event for endpoint {request.url} failed to send: {exc}"
+            )
+            return response
+
+        return response
+    except Exception as e:
+        logger.warning("exception caught")
+        prepare_and_log_request(
+            endpoint, e.args[0], now, fides_source, e.__class__.__name__
```

Comment:
I don't think an exception is guaranteed to have an argument - in which case e.args[0] would throw an indexerror or similar. You also have status_code being an `int` below, but that's not guaranteed for this first argument to an exception.

Reply:
Thanks, I'll check this out. 

This is also a rare exception clause that will catch if, for some reason, a non-HTTP exception is thrown from our endpoint that has a status code of <400. I don't expect this will ever happen, but may be good to have in case.

Reply:
I guess technically a non-HTTP exception doesn't have a status code necessarily, and if it does there's no guarantee where status code would be. What do you think of just hard-coding `500` or something here?

Reply:
well python exceptions don't have codes, the first argument that you're displaying above `e.args[0]` is usually the message.  For example, if I raised Exception("Bad code"), - "Bad code" is what your currently sending in as your status code.

Reply:
500 is fine because Python Exceptions don't have a code, but I'd think the actual message would be meaningful too, the fake 500 doesn't tell you much.